python.d/monit.chart.py

```diff
@@ -0,0 +1,146 @@
+        'lines': []
+    },
+    'file': {
+        'options': ['files', 'Files', 'files', 'filesystem', 'monit.files', 'line'],
+        'lines': []
+    },
+    'fifo': {
+        'options': ['fifos', 'Pipes (fifo)', 'pipes', 'filesystem', 'monit.fifos', 'line'],
+        'lines': []
+    },
+    'program': {
+        'options': ['programs', 'Programs statuses', 'programs', 'applications', 'monit.programs', 'line'],
+        'lines': []
+    },
+    'process': {
+        'options': ['processes', 'Processes statuses', 'processes', 'applications', 'monit.services', 'line'],
+        'lines': []
+    },
+    'process_uptime': {
+        'options': ['processes uptime', 'Processes uptime', 'seconds', 'applications', 'monit.process_uptime', 'line', 'hidden'],
+        'lines': []
+    },
+    'process_threads': {
+        'options': ['processes threads', 'Processes threads', 'threads', 'applications', 'monit.process_threads', 'line'],
+        'lines': []
+    },
+    'process_children': {
+        'options': ['processes childrens', 'Child processes', 'childrens', 'applications', 'monit.process_childrens', 'line'],
+        'lines': []
+    },
+    'host': {
+        'options': ['hosts', 'Hosts', 'hosts', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'host_latency': {
+        'options': ['hosts latency', 'Hosts latency', 'milliseconds/s', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'net': {
+        'options': ['interfaces', 'Network interfaces and addresses', 'interfaces', 'network', 'monit.networks', 'line'],
+        'lines': []
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', "http://127.0.0.1:2812/_status?format=xml&level=full")
+        self.order = ORDER
+        self.definitions = CHARTS
+
+    def check(self):
+        self._manager = self._build_manager()
+        raw_data = self._get_raw_data()
+        if raw_data:
+            try:
+                xml = ET.fromstring(raw_data)
+            except ET.ParseError:
+                self.debug('%s is not a vaild XML page. Please add "_status?format=xml&level=full" to monit URL.' % self.url)
+                return None
+            return True
+        return None
+
+    def _get_data(self):
+        data = None
+        raw_data = self._get_raw_data()
+        if raw_data:
+            try:
+                xml = ET.fromstring(raw_data)
+            except ET.ParseError:
+                self.debug('%s is not a vaild XML page. Please add "_status?format=xml&level=full" to monit URL.' % self.url)
+                return None
+            data = {}
+
+            for svc_id in DEFAULT_SERVICES_IDS:
+                if svc_id not in DEFAULT_SERVICES_IDS:
+                    continue
```

Comment:
This doesn't make any sense

Reply:
Yep, it's nonsense, you're right. It was supposed to be configured via conf file. For example, if user don't want to export all checks. I can just remove it for now and implement it later. Ok?

Reply:
I mean
```for svc_id in DEFAULT_SERVICES_IDS```
```if svc_id not in DEFAULT_SERVICES_IDS```

Same `DEFAULT_SERVICES_IDS`, so ` svc_id` is always in `DEFAULT_SERVICES_IDS`

Reply:
> Same DEFAULT_SERVICES_IDS, so svc_id is always in DEFAULT_SERVICES_IDS

It's exactly what i meant. For now i'll just delete 106-107. Ok?

Reply:
Ok

Reply:
Removed